source/_posts/2015-10-09-the-slack-users-code.markdown

```diff
@@ -0,0 +1,258 @@
+
+
+#### Stay on Topic
+
+Try to stay on topic. The fewer people in a channel, the less important this is,
+but still. If people are off topic and annoying you, you can tell them to
+":goto: #watercooler" (or whatever appropriate channel, and you should totally
+[alias the :arrow_right: emoji as :goto:](https://my.slack.com/customize/emoji)).
+
+
+#### Invite People to Your Channel
+
+If the #some-product engineers are having a problem with deployment and they
+think it might be a problem that the #infrastructure folks could help solve,
+here's how I recommend they should handle it. Once they make that determination
+(I assume they've been talking about the problem in #some-product for a while,
+now), they should elect a single one of them to pop into #infrastructure.
+
+That person should see if anyone else has already brought the problem up and, if
+so, follow that conversation (possibly to another channel, as we'll see
+shortly). If not, then they should say something like, "@infrateamlead: We're
+having a problem with deployment over in #some-product. Does anyone have time to
+pop over and help us work it out?"
+
+You ask the team lead because, as with any other task you want to hand to
+another team, you don't jump the chain of command: you let the leader decide
+who's interruptable or appropriate for the task. You mention what channel you've
+been talking about the problem in so that the nominated helper knows where to
+go, _as well as anyone else who wanders into #infrastructure with the same
+problem_.
+
+Then whoever it is from #infrastructure that's going to help can join
+ #some-product and see all the history of the discussion about the problem and
+get caught up.  And, you know, then hopefully help out in fixing the problem.
+This protocol or something like it is especially important for heavily
+interrupt-driven teams like Ops, IT, Internal Tools, etc.
+
+
+### Off-Topic Guidelines
+
+#### Name it After the Topic
+
+Off-Topic channels still have topics, they're just not work-required topics.
+This can be something very general like #games or very specific like
+ #long-distance-trail-running (but remember, Don't Create Extraneous Channels).
+Try to keep in mind the geographically distributed nature of your organization.
+If you're making a channel to find people to play cards with at lunch, maybe put
+the office or city name in the channel title, as well.
+
+
+#### Name it After the Role
+
+There is a special kind of off-topic channel that bears calling out. You
+probably want to have a channel for Software Engineers to hang out and talk
+about their jobs. Or Engineering Managers or whatever. These are related to the
+work that people are doing, but they're probably not required to be in the
+channel
+in order to get their job done. These are hubs for people to do frequent, low
+level professional development. They should be grouped around _roles_ not
+_titles_. Whether you call it #software-engineering or #programming is a bit of
```

Comment:
I’m interested to hear more about why role grouping is better than titles.

I think I see why when you get into `#product` instead of `#product-managers`, and a bit about inclusiveness around everyone who has a stake or interest in the topic, rather than only the people of the named role.